src/Ledger.cs

```diff
@@ -215,4 +215,51 @@ public static async Task DeleteOneAsLedger<T>(this IMongoCollection<T> collectio
             throw;
         }
     }
-}
\ No newline at end of file
+
+    /// <summary>
+    /// Verifies if an item's history is unchanged by comparing the current hash with the stored hash
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="collection"></param>
+    /// <param name="document"></param>
+    /// <returns></returns>
+    public static async Task<bool> VerifyOneFromLedger<T>(this IMongoCollection<T> collection, T document)
+    {
+        // Get the database and log collection
+        var db = collection.Database.Client.GetDatabase(collection.Database.DatabaseNamespace.DatabaseName);
+        var logCollection = db.GetCollection<LogRecord<T>>($"{collection.CollectionNamespace.CollectionName}{LOG_COLLECTION_PREFIX}");
+
+        // Get the original ID of the entity
+        var originalId = document.ToBsonDocument()["_id"].ToString();
```

Comment:
What if document is "null", there should be also some null check, if it is null some exception should be thrown.